test/scheduler_test.rb

```diff
@@ -456,6 +456,52 @@
            /#{Resque::Scheduler.send(:internal_name)}: cage/
   end
 
+  test 'gracefully shuts down rufus-scheduler threads' do
+    class BeforeEnqueueJob
+      @queue = :quick
+
+      def self.before_enqueue_example(*)
+        if Resque.redis.setnx('before_enqueue_job:enqueued', 'true')
+          sleep 5 # sleep longer than test timeout
+          true
+        else
+          false # enqueue only once
+        end
+      end
+
+      def self.perform(_, _)
+      end
+    end
+
+    schedule = {
+      'BeforeEnqueueJob' => { cron: '* * * * * *', class: 'BeforeEnqueueJob' }
+    }
+
+    Resque::Scheduler.enqueue_from_config(schedule['BeforeEnqueueJob'])
+
+    pid = fork do
+      Resque.schedule = schedule
+      Resque::Scheduler.run
+    end
+
+    begin
+      Timeout.timeout(3) do
+        loop do
+          break if Resque.redis.get('before_enqueue_job:enqueued') == 'true'
```

Comment:
not a huge fan of how this uses the internals of the job.. maybe there could be a `BeforeEnqueueJob.enqueue_started?` helper method? that would make the test easier maybe